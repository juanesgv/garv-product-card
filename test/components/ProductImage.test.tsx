import * as React from 'react';
import rendered from 'react-test-renderer'
import { ProductImage, ProductCard } from '../../src/components';
import { product2 } from '../data/products';

describe('ProductImage', ()=>{

    test('Debe mostrar el componente correctamente con la imagen personalizado', ()=>{
        const wrapper = rendered.create(
            <ProductImage img='https://hola.jpg'/>
        )

        expect(wrapper.toJSON()).toMatchSnapshot()
    })

    test('debe de mostrar el componente la imagen del producto', () => {
        const wrapper = rendered.create(
            <ProductCard product={product2}>
                {
                    ()=> (
                        <ProductImage />
                    )
                }
            </ProductCard>
        )

        expect(wrapper.toJSON()).toMatchSnapshot()
    })
    
})
