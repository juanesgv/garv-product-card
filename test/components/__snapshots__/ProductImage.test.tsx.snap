// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Debe mostrar el componente correctamente con la imagen personalizado 1`] = `
<img
  alt="Product"
  className="productImg undefined"
  src="https://hola.jpg"
/>
`;

exports[`ProductImage debe de mostrar el componente la imagen del producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="Producto.jpg"
  />
</div>
`;
